object/slicer/slicer.go

```diff
@@ -355,14 +355,14 @@ func (x *PayloadWriter) Write(chunk []byte) (int, error) {
 
 	if !x.withSplit {
 		x.splitID = object.NewSplitID()
+		x.withSplit = true
```

Comment:
if code movement matters, and this is not obvious (`x.withSplit` isn't used in the current condition block directly, maybe inside calling methods only), i'd write a comment about this: `note: don't move, <reason>`

Reply:
Make sense, added comment about reason